src/App.tsx

```diff
@@ -15,48 +20,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.NONE:
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, reverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const isResetButton = sortType !== SortType.NONE || isReversed;
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    { isReversed, sortType },
+  );
+  const reset = () => {
+    reverse(false);
+    setSortType(SortType.NONE);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            { 'is-light': sortType !== SortType.ALPHABET },
```

Comment:
use variable for this expression

Reply:
But it's a simple expression. It is essential to make variables for this type of simple condition
?